### PR TITLE
Expose default values for native functions/methods

### PR DESCRIPTION
**Pull request by rvanvelzen**
This should help getting #1244 over the finish line

### CODE REVIEW

File: tests/PHPStan/Analyser/data/native-reflection-default-values.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace NativeReflectionDefaultValues;
+
+use function PHPStan\Testing\assertType;
+
+function () {
+	assertType('ArrayObject<*NEVER*, *NEVER*>', new \ArrayObject());
```
Comment (rvanvelzen) [inline tests/PHPStan/Analyser/data/native-reflection-default-values.php:8]: This isn't ideal but matches the behavior with the actual default value passed (which is the next line)

**Comment by rvanvelzen** [comment]
Hmm, obviously this won't work for PHP<8 :/

**Comment by ondrejmirtes** [comment]
Oh, you have no idea what you’ve signed up for 😊 This will unblock multiple nice things but I have to do sime preliminary work first and then I’ll get back to you 😊

**Comment by ondrejmirtes** [comment]
FYI @rvanvelzen I'm aware of your many PRs you sent in the last 24 hours, but I first want to finish my promise of the work needed for this area, and then I'll look at the other PRs :)

**Comment by rvanvelzen** [comment]
@ondrejmirtes no worries, it gives me enough time to work on implementing `assert` tags :)

**Comment by ondrejmirtes** [comment]
Alright, doing as promised, here's the description of what I want from you: https://github.com/phpstan/phpstan-src/pull/1276#issuecomment-1116002090 Thanks :)

**Comment by ondrejmirtes** [comment]
Superseded by https://github.com/phpstan/phpstan-src/pull/1300

**Pull request closed without merging**